packages/ui/src/proposals/types/proposalsDetailed.ts

```diff
@@ -0,0 +1,18 @@
+import { asBlock, Block } from '@/common/types'
+import { ProposalDetailedFieldsFragment, VoteFieldsFragment } from '@/proposals/queries'
+
+import { asProposal, Proposal } from './proposals'
+
+export interface ProposalDetailed extends Proposal {
```

Comment:
Maybe we should go with `ProposalWithDetails` just because we already have `MemberWithDetails` and `WorkerWithDetails`.

The benefit of the `FooWithDetails` is that you can write `Foo...` and see what other forms are there with autocomplete (usually that'd be `Foo` and `FooWithDetails`. Also the cast functions names have the same order as the base object, e.g.: `asFooWithDetails()` vs `asDetailedFoo` (for `FooDetailed`).